server/src/test/java/org/elasticsearch/health/node/HealthInfoTests.java

```diff
@@ -23,60 +22,34 @@ protected Writeable.Reader<HealthInfo> instanceReader() {
-            DiskHealthInfo diskHealthInfo = randomBoolean()
-                ? new DiskHealthInfo(randomFrom(HealthStatus.values()))
-                : new DiskHealthInfo(randomFrom(HealthStatus.values()), randomFrom(DiskHealthInfo.Cause.values()));
-            diskInfoByNode.put(randomAlphaOfLengthBetween(10, 100), diskHealthInfo);
-        }
-        return new HealthInfo(diskInfoByNode, randomBoolean() ? randomDslHealthInfo() : null);
+        var diskInfoByNode = randomMap(0, 10, () -> tuple(randomAlphaOfLength(10), randomDiskHealthInfo()));
+        var repositoriesInfoByNode = randomMap(0, 10, () -> tuple(randomAlphaOfLength(10), randomRepoHealthInfo()));
+        return new HealthInfo(diskInfoByNode, randomBoolean() ? randomDslHealthInfo() : null, repositoriesInfoByNode);
     }
 
     @Override
     public HealthInfo mutateInstance(HealthInfo originalHealthInfo) {
-        Map<String, DiskHealthInfo> diskHealthInfoMap = originalHealthInfo.diskInfoByNode();
-        Map<String, DiskHealthInfo> diskHealthInfoMapCopy = new HashMap<>(diskHealthInfoMap);
-        if (diskHealthInfoMap.isEmpty()) {
-            diskHealthInfoMapCopy.put(
-                randomAlphaOfLength(10),
-                new DiskHealthInfo(randomFrom(HealthStatus.values()), randomFrom(DiskHealthInfo.Cause.values()))
-            );
-        } else {
-            switch (randomIntBetween(1, 3)) {
-                case 1 -> {
-                    diskHealthInfoMapCopy.put(
-                        randomAlphaOfLength(10),
-                        new DiskHealthInfo(randomFrom(HealthStatus.values()), randomFrom(DiskHealthInfo.Cause.values()))
-                    );
-                }
-                case 2 -> {
-                    String someNode = randomFrom(diskHealthInfoMap.keySet());
-                    diskHealthInfoMapCopy.put(
-                        someNode,
-                        new DiskHealthInfo(
-                            randomValueOtherThan(diskHealthInfoMap.get(someNode).healthStatus(), () -> randomFrom(HealthStatus.values())),
-                            randomFrom(DiskHealthInfo.Cause.values())
-                        )
-                    );
-                }
-                case 3 -> {
-                    diskHealthInfoMapCopy.remove(randomFrom(diskHealthInfoMapCopy.keySet()));
-                }
-                default -> throw new IllegalStateException();
-            }
-        }
         return new HealthInfo(
-            diskHealthInfoMapCopy,
-            randomValueOtherThan(originalHealthInfo.dslHealthInfo(), HealthInfoTests::randomDslHealthInfo)
+            mutateMap(originalHealthInfo.diskInfoByNode(), () -> randomAlphaOfLength(10), HealthInfoTests::randomDiskHealthInfo),
+            randomValueOtherThan(originalHealthInfo.dslHealthInfo(), HealthInfoTests::randomDslHealthInfo),
+            mutateMap(originalHealthInfo.repositoriesInfoByNode(), () -> randomAlphaOfLength(10), HealthInfoTests::randomRepoHealthInfo)
```

Comment:
We try to narrow down the mutation to one element at a time and not all. Mutating everything might hide that we missed a check in the equality for example. Can you restructuring in a way that it changes only one element during a call.

Reply:
Done, [98db870](https://github.com/elastic/elasticsearch/pull/104614/commits/98db870272c39a741ec3e5e4b2b9674912214bcc), please check if this works for you.